lib/iris/fileformats/netcdf.py

```diff
@@ -3206,3 +3245,114 @@ def is_valid_packspec(p):
+            dimension, e.g. {'time': 10, 'model_levels':1}.
+
+        Notes
+        -----
+        This function acts as a contextmanager, for use in a 'with' block.
+
+        Example:
+
+        >>> from iris.fileformats.netcdf import CHUNK_CONTROL
+        >>> with CHUNK_CONTROL.set('var1', model_level=1, time=50):
+        ...     cubes = iris.load(filename)
+
+        When ``var_names`` is present, the chunksize adjustments are applied
+        only to the selected variables.  However, for a CF data variable, this
+        extends to all components of the (raw) cube created from it.
+
+        **Un**-adjusted dimensions have chunk sizes set in the 'usual' way.
+        That is, according to the normal behaviour of
+        :meth:`iris._lazy_data.as_lazy_data`, which is : chunksize is based on
+        the file variable chunking, or full variable shape; this is scaled up
+        or down by integer factors to best match the Dask "default chunksize",
+        i.e. the setting configured by
+        ``dask.config.set({'array.chunk-size': '250MiB'})``.
+
+        """
+        old_settings = deepcopy(self.var_dim_chunksizes)
+        if var_names is None:
+            var_names = ["*"]
+        elif isinstance(var_names, str):
+            var_names = [var_names]
+        try:
+            for var_name in var_names:
+                # Note: here we simply treat '*' as another name.
+                # A specific name match should override a '*' setting, but
+                # that is implemented elsewhere.
+                if not isinstance(var_name, str):
+                    msg = (
+                        "'var_names' should be an iterable of strings, "
+                        f"not {var_names!r}."
+                    )
+                    raise ValueError(msg)
+                dim_chunks = self.var_dim_chunksizes.setdefault(var_name, {})
+                for dim_name, chunksize in dimension_chunksizes.items():
+                    if not (
+                        isinstance(dim_name, str)
+                        and isinstance(chunksize, int)
+                    ):
+                        msg = (
+                            "'dimension_chunksizes' kwargs should be an iterable "
+                            f"of (str, int) pairs, not {dimension_chunksizes!r}."
+                        )
+                        raise ValueError(msg)
+                    dim_chunks[dim_name] = chunksize
+            yield
+        finally:
+            self.var_dim_chunksizes = old_settings
+
+
+# Note: the CHUNK_CONTROL object controls chunk sizing in the
+# :meth:`_get_cf_var_data` method.
+# N.B. :meth:`_load_cube` also modifies this when loading each cube,
+# introducing an additional context in which any cube-specific settings are
+# 'promoted' into being global ones.
+
+#: A :class:`ChunkControl` object providing user-control of Dask chunking
+#: when Iris loads netcdf files.
+CHUNK_CONTROL: ChunkControl = ChunkControl()
```

Comment:
Nitpick: I would go with lower case here, `CHUNK_CONTROL.set(...)` looks a bit shouty.

Reply:
Isn't it mostly for simple things with "fixed" value, rather than objects with complex behaviour like context managers?

Reply:
Oh. In that case I've messed up elsewhere! Look forward to seeing what @pp-mo has to say 😅

Reply:
I'm going to do a drive-by nitpick as I'm getting all the email alerts for this thread in my inbox :)

There should be no need for the type hint here, Python type hinter should be smart enough to know that 

`CHUNK_CONTROL = ChunkControl()`

is indeed a `ChunkControl` instance.

Reply:
>  @TomekTrzeciak Nitpick

As regards captilisation, I was basically mimicking [iris.FUTURE](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris.html#iris.FUTURE).
Which makes _some_ kind of sense IMHO as this is very much the same style of thing.

On the other hand it's possible that, when that was introduced, we were slightly conflicted about the difference between a "constant value" and a "global variable", as you suggest, since in Python there is essentially no actual distinction.

----
> @jamesp drive-by nitpick

It wasn't to enable type hinting, but trying to inject type into the docstring representation.  
https://pp-iris.readthedocs.io/en/nc_chunk_control/generated/api/iris/fileformats/netcdf.html#iris.fileformats.netcdf.CHUNK_CONTROL
Hence the colon.  But it doesn't seem to actually do so, presumably because Sphinx is not so configured 😞